src/luascript.cpp

```diff
@@ -8090,11 +8090,15 @@ int LuaScriptInterface::luaPlayerSetGuildLevel(lua_State* L)
 	uint8_t level = getNumber<uint8_t>(L, 2);
 	Player* player = getUserdata<Player>(L, 1);
 	if (player) {
-		player->setGuildLevel(level);
-		pushBoolean(L, true);
-	} else {
-		lua_pushnil(L);
+		if (const Guild* guild = player->getGuild()) {
+			if (const GuildRank* rank = guild->getRankByLevel(level)) {
+				player->setGuildRank(rank);
+				pushBoolean(L, true);
+				return 1;
+			}
+		}
 	}
+	pushBoolean(L, false);
```

Comment:
Fixed, I guess. Now pushes nil if either player is null or player guild is null, and pushes false if there's no rank available.